specificatie/datum.yaml

```diff
@@ -0,0 +1,163 @@
+                example:
+                  [
+                    { "type": "GbaDatum", "waarde": "20180701" },
+                    { "type": "GbaDatum", "waarde": "20161200" },
+                    { "type": "GbaDatum", "waarde": "20100000" },
+                    { "type": "GbaDatum", "waarde": "00000000" },
+                    { "type": "Datum", "waarde": "2018-07-01" },
+                    { "type": "JaarMaandDatum", "jaar": "2016", "maand": "12" },
+                    { "type": "JaarDatum", "jaar": "2010" },
+                    { "type": "OnbekendDatum", "onbekend": true }
+                  ]
+components:
+  schemas:
+    AbstractDatum:
+      type: object
+      required:
+        - type
+      properties:
+        type:
+          type: string
+      discriminator:
+        propertyName: type
+        mapping:
+          GbaDatum: '#/components/schemas/GbaDatum'
+          Datum: '#/components/schemas/Datum'
+          OnbekendDatum: '#/components/schemas/OnbekendDatum'
+          JaarDatum: '#/components/schemas/JaarDatum'
+          JaarMaandDatum: '#/components/schemas/JaarMaandDatum'
+    GbaDatum:
+      description: Datum conform LO GBA
+      allOf:
+        - $ref: '#/components/schemas/AbstractDatum'
+        - type: object
+          properties:
+            waarde:
+              type: string
+              pattern: "^[0-9]{8}$"
+              minLength: 8
+              maxLength: 8
+      required:
+        - waarde
+      example: 20210801
+    Datum:
+      description: Datum conform iso8601
+      allOf:
+        - $ref: '#/components/schemas/AbstractDatum'
+        - type: object
+          properties:
+            waarde:
+              type: string
+              format: date
+              minLength: 10
+              maxLength: 10
+      required:
+        - waarde
+      example: 2021-08-01
+    OnbekendDatum:
+      description: representatie voor een onbekend datum
+      allOf:
+        - $ref: '#/components/schemas/AbstractDatum'
+        - type: object
+          properties:
+            onbekend:
+              type: boolean
+              default: true
+      required:
+        - onbekend
+    JaarDatum:
+      description: representatie voor een datum waarvan maand en dag onbekend zijn
+      allOf:
+        - $ref: '#/components/schemas/AbstractDatum'
+        - type: object
+          properties:
+            jaar:
+              $ref: '#/components/schemas/Jaar'
+      required:
+        - jaar
+    JaarMaandDatum:
+      description: representatie voor een datum waarvan de dag onbekend is
+      allOf:
+        - $ref: '#/components/schemas/AbstractDatum'
+        - type: object
+          properties:
+            jaar:
+              $ref: '#/components/schemas/Jaar'
+            maand:
+              $ref: '#/components/schemas/Maand'
+      required:
+        - jaar
+        - maand
+    Jaar:
+      type: integer
+      format: int32
+      minimum: 0
```

Comment:
De waarde kan niet 0 zijn, want dan is de GBA datum 00000000, oftewel volledig onbekend

Reply:
fixed